unison-runtime/src/Unison/Runtime/Machine.hs

```diff
@@ -720,11 +748,31 @@ apply env henv !activeThreads !stk !k !ck !args !val =
           stk <- discardFrame stk
           stk <- bump stk
           poke stk v
-          yield env henv activeThreads stk k
+          yield yld env henv activeThreads stk k
       | otherwise = die $ "applying non-function: " ++ show v
+{-# INLINE apply' #-}
+
+apply ::
+  Int ->
+  CCache ->
+  HEnv ->
+  ActiveThreads ->
+  Stack ->
+  K ->
+  Bool ->
+  Args ->
+  Val ->
+  IO ()
+apply !yld env henv !activeThreads !stk !k !ck !args !val
+  | yld <= 0 = do
+      CNC.yield
+      apply' yieldSteps env henv activeThreads stk k ck args val
+  | otherwise =
+      apply' (yld - 1) env henv activeThreads stk k ck args val
```

Comment:
Does it make any difference to swap these around? Put the most common case first.

```suggestion
  | yld > 0 =  apply' (yld - 1) env henv activeThreads stk k ck args val
  | otherwise = do
      CNC.yield
      apply' yieldSteps env henv activeThreads stk k ck args val
```

Reply:
It doesn't seem to make any difference.

What does make a difference is turning it into something like:

```
do
  yld <- if yld > 0
           then pure $ yld-1
           else yieldSteps <$ CNC.yield
  apply' yld ...
```

which is slower for some reason.